### ISSUE TITLE
Contacts Picker UI should have a "Select All" option 

### ISSUE THREAD

**Issue by adityapunjani** [beverloo/contact-api#15]
I think without having a select all option, the contacts picker is going to be futile for most use cases. I believe it is imperative we allow users to select all contacts unambiguously with the right permission prompt of-course (and unselect those that the user wants to opt-out) as the contacts import flow is usually a one-time action, part of an on-boarding flow or discover people flow in most social apps (Twitter, Pinterest, Facebook & Instagram that I build)  

Without having a select all option, this API adds no value to improving UX and helping users find the right people to connect with. I would also suggest a default mode all contacts pre-selected, behind a strongly worded permission prompt. The goal here is to reduce the number of inputs that the user has to make to achieve the ideal result. 

In the context of using the contact picker in an email application - I think most email apps will do a better job auto-completing than having to select individual contact info via a picker. The feels more like finding a solution for a problem that doesn't exist :) Even here the core use case would be to import all contacts and then optimize autocomplete. 

Thoughts?

**Comment by laukstein** [comment]
I agree,

**Comment by rayankans** [comment]
@adityapunjani I agree it would be useful, but it is not critical for the feature to function, or critical for privacy/security reasons. The spec shouldn't enforce a `Select All` option, it should probably just recommend it, and allow this to be a user agent decision.

**Comment by aarongustafson** [comment]
I agree with @rayankans as it gives us the flexibility to test it out and remove it if necessary.

**Comment by comp615** [comment]
Following up here, we've been discussing this internally at Twitter and believe it may be a big barrier to usefulness to the point where we might not use it. I understand many of the use cases are looking for a single contact, or are going to actively send messages or spam anyone selected, and I definitely do not want that to happen on the internet...but for Twitter, being able to connect to as many people as they know is a user convenience. Obviously we have no way to say, hey we are going to use the data in a certain way...so I get the conundrum.

Would it be possible to have the option but have it not be selectable it by default? Or have a process to verify data storage / usage to whitelist? Or per-site-rank? Do you have any other suggestions for how we can help people find everyone in their book?

**Comment by rayankans** [comment]
Thanks for the feedback @comp615. You make some good points, but I really don't think enforcing something like that belongs in the spec. As I said in a comment above, UI sections in specs just enforce things critical to user security/privacy, which allows for flexibility when browsers implementers develop their own UI.

I'll keep the issue open so browser vendors can have this developer feedback for when they make progress on the contacts api.
